Make network timeout configurable via environment variable

kolanos opened the pull request:
Closes #138

coreylight commented:
It looks like js and java both use millis for this var and timeoutWindow. Can we use millis for the standardization? If we have to make a breaking change it might be worth it to do it now. We could at least append `_millis` to the var and have two vars, deprecate the old ones, and when we want to make a breaking change down the line, remove and rename?

kolanos commented:
@coreylight It's seconds because that's what `requests` and `signal` accept. I think this falls under language idioms, as I'm not aware of any Python library that accepts timeouts in milliseconds. Instead they would expect milliseconds to be expressed as a float (i.e. `0.5` == `500` for the timeout window. Also, I'd just be `value / 1000`. The only thing I can think of where we might want to break from idioms is if someone is running multiple runtimes and expecting all the environment varaibles to be the same unit. But I suspect that's pretty rare.

@ewindisch Thoughts?

kolanos commented:
Updated to accept `network_timeout` and `timeout_window` as millisecond integers. Added warnings for anyone passing a float value via environment variable.
